src/EditorFeatures/Core/ExternalAccess/VSTypeScript/IVSTypeScriptEditorInlineRenameService.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript
+{
+    internal readonly struct VSTypeScriptInlineRenameLocation
+    {
+        public Document Document { get; }
+        public TextSpan TextSpan { get; }
+
+        public VSTypeScriptInlineRenameLocation(Document document, TextSpan textSpan)
+        {
+            this.Document = document;
+            this.TextSpan = textSpan;
+        }
+    }
+
+    internal enum VSTypeScriptInlineRenameReplacementKind
+    {
+        NoConflict,
+        ResolvedReferenceConflict,
+        ResolvedNonReferenceConflict,
+        UnresolvedConflict,
+        Complexified,
+    }
+
+    internal readonly struct VSTypeScriptInlineRenameReplacement
```

Comment:
>VSTypeScriptInlineRenameReplacement [](start = 29, length = 35)

It seems F# is not following the patterns we have elsewhere. For simple wrappers we have been using pattern like so:

```C#
    internal readonly struct VSTypeScriptInlineRenameReplacementWrapper
    {
        private readonly InlineRenameReplacement _underlyingObject;

        public VSTypeScriptInlineRenameReplacementWrapper(InlineRenameReplacement underlyingObject)
        {
           _underlyingObject = underlyingObject;
        }

        public VSTypeScriptInlineRenameReplacementKind Kind => _underlyingObject.Kind;
        public TextSpan OriginalSpan => _underlyingObject.OriginalSpan;
        public TextSpan NewSpan => _underlyingObject.NewSpan;
    }
```